docs/design/byos.md

```diff
@@ -1,13 +1,142 @@
+In return, customers must give up control over tools, capacity, and speed.
+On the other end of the spectrum, private agents offer the exact opposite set of trade-offs: full control over everything, at the expense of maintaining and paying for infrastructure.
+With private agents, elasticity is difficult to achieve.
 
-## State
+Bring Your Own Subscription (BYOS) represents a middle ground:
+it pairs the convenience and elastic capacity of the hosted pool with the control and flexibility of private agents.
+**Azure Pipelines will manage a set of build/release agents to the customer's specification, completely automated, in the customer's Azure subscription.**
 
-This is in the early design phase and we are looking for feedback.  Feedback can be provided as issues in this repo with the enhancement and design tags.
+## State of this spec
 
+This is in the early design phase and we are looking for feedback in the PR (or as issues in the repo).
+
+## Customer scenarios
+
+The theme throughout these scenarios is that the customer wants hosted elasticity but customization beyond what Hosted offers.
+
+General themes are around:
+- VM specs (memory, CPU, disk) and network environment
+- Preinstalled software
+- Agent reuse
+
+Related, on-premises customers have often asked us for access to the Hosted pools.
+Pools using this feature should be available to Azure DevOps Server customers.
+By making our hosted images available in the Azure gallery, customers get an equivalent to the Hosted pools without having to worry about multi-tenancy, a new billing relationship, etc.
+
+### VM specs and environment
+
+1. Customer wants more memory, more processor, or more IO than our native images.
+2. [Customer](https://github.com/MicrosoftDocs/vsts-docs/issues/2985) wants an NCv2 VM with particular instruction sets for machine learning. (It's niche enough that we won't stand up dedicated hosted pools, but broad enough to be very interesting for our business.)
+3. Customer wants additional storage attached to the VM. *(Real scenario from a medium-sized customer)*
+4. Customer wants to deploy to a private App Service. It's in a private VNET with no inbound connectivity. Today, this customer is forced to keep private agents standing by.
+
+### Preinstalled software
+
+1. Customer wants Windows 2016 with VS 2019 Preview. (We only offer certain combos like Win2016 + VS2017 and Win2019 + VS2019.)
+2. Customer wants to pin a specific set of tools and dependencies, preconfigured on the image.
+3. Customer wants extreme control over the exact OS build, settings, versions, and so on.
+
+### Agent reuse
+
+1. Customer wants to run several consecutive jobs on an agent to take advantage of things like incremental source and machine-level package caches. But, they don't want to run unnecessary VMs overnight when there's no load. They want to specify minimum and maximum # of agents associated with time ranges, but reuse agents within that span.
```

Comment:
Two factors here: 1 is cost savings. The other is "every hundredth build, whack this agent"